### PR TITLE
feat: Details implementation

### PR DESCRIPTION
**Pull request by aballerr**
Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary), and should start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- docs: Documentation only changes
- feat: A new feature
- fix: A bug fix
- perf: A code change that improves performance
- refactor: A code change that neither fixes a bug nor adds a feature
- style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- test: Adding missing tests or correcting existing tests

Example commit messages:

- feat: adds support for gnosis safe wallet
- fix: removes a polling memory leak
- chore: bumps redux version

Other things to note:

- Please describe the change using verb statements (ex: Removes X from Y)
- PRs with multiple changes should use a list of verb statements
- Add any relevant unit / integration tests
- Changes will be previewable via vercel. Non-obvious changes should include instructions for how to reproduce them

### CODE REVIEW

File: src/nft/types/collection/collection.ts [hunk 1]
```diff
@@ -49,10 +49,10 @@ export enum ActivityEventType {
...
 
 export enum ActivityEventTypeDisplay {
-  'LISTING' = 'Listing',
-  'SALE' = 'Sale',
-  'TRANSFER' = 'Transfer',
-  'CANCEL_LISTING' = 'Cancel',
+  'LISTING' = 'Listed',
+  'SALE' = 'Sold',
+  'TRANSFER' = 'Transferred',
+  'CANCEL_LISTING' = 'Cancellation',
```
Comment (aballerr) [inline src/nft/types/collection/collection.ts:55]: I was told this should be "Cancellation"

File: src/nft/components/details/AssetActivity.tsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    margin-top: 40px;
+  }
+  ::-webkit-scrollbar-thumb {
+    background: ${({ theme }) => theme.backgroundOutline};
+    border-radius: 8px;
+  }
+`
+
+const EventCellContainer = styled.div`
+  margin-top: 5px;
```
Comment (aballerr) [inline src/nft/components/details/AssetActivity.tsx:121]: This is to align the text in the table.  Not 100% why it isn't aligned, but that's the reason for the random 5px

File: src/nft/components/details/DetailsContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    <Details>
+      <GridItem
+        header="Contract address"
+        body={
+          <Center onClick={copy}>
+            {shortenAddress(address, 2, 4)} <CopyIcon size={13} />
+          </Center>
+        }
+      />
+      <GridItem header="Token ID" body={tokenId.length > 9 ? stringShortener(tokenId) : tokenId} />
```
Comment (aballerr) [inline src/nft/components/details/DetailsContainer.tsx:91]: edge case where token can be super long

File: src/nft/hooks/useBag.ts [hunk 4]
```diff
@@ -57,6 +58,10 @@ export const useBag = create<BagState>()(
         set(({ bagExpanded }) => ({
           bagExpanded: !bagExpanded,
         })),
+      openBag: () =>
+        set(() => ({
+          bagExpanded: true,
```
Comment (aballerr) [inline src/nft/hooks/useBag.ts:63]: this is for when a user clicks "buy".  The only other option was toggle

File: src/nft/utils/fetchPrice.ts [hunk 5]
```diff
@@ -20,5 +20,5 @@ export const fetchPrice = async (currency: Currency = Currency.ETH): Promise<num
 
 export function useUsdPrice(asset: GenieAsset): string {
   const { data: fetchedPriceData } = useQuery(['fetchPrice', {}], () => fetchPrice(), {})
-  return fetchedPriceData ? (parseFloat(formatEther(asset.priceInfo.ETHPrice)) * fetchedPriceData).toString() : ''
+  return fetchedPriceData ? (parseFloat(formatEther(asset?.priceInfo?.ETHPrice)) * fetchedPriceData).toString() : ''
```
Comment (JackShort) [inline src/nft/utils/fetchPrice.ts]: when an asset isn't listed for sale this causes an error check if the asset has a price first / is available

File: src/nft/components/details/DetailsContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+const GridItem = ({ header, body }: { header: string; body: React.ReactNode }) => {
+  return (
+    <div>
+      <Header>{header}</Header>
+      <Body>{body}</Body>
+    </div>
+  )
+}
+
+const stringShortener = (text: string) => `${text.substring(0, 4)}...${text.substring(text.length - 4, text.length)}`
```
Comment (JackShort) [inline src/nft/components/details/DetailsContainer.tsx:70]: can you put this in the utils file with `shortenAddress`

File: src/nft/components/details/DetailsContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+const DetailsContainer = ({ asset, collection }: { asset: GenieAsset; collection: CollectionInfoForAsset }) => {
+  const { address, tokenId, tokenType, creator } = asset
+  const { totalSupply } = collection
+
+  const [, setCopied] = useCopyClipboard()
+  const copy = useCallback(() => {
+    setCopied(address || '')
```
Comment (JackShort) [inline src/nft/components/details/DetailsContainer.tsx:78]: use `??` its better than `||` for nullish checks

File: src/nft/components/details/InfoContainer.tsx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+const InfoContainer = ({
+  children,
+  primaryHeader,
+  secondaryHeader,
+  defaulOpen,
+}: {
+  children: JSX.Element
+  primaryHeader: string
+  secondaryHeader: React.ReactNode
+  defaulOpen?: boolean
```
Comment (JackShort) [inline src/nft/components/details/InfoContainer.tsx:72]: typo should be `defaultOpen`

File: src/nft/components/icons.tsx [hunk 9]
```diff
@@ -1495,11 +1495,10 @@ export const EmptyNFTWalletIcon = (props: SVGProps) => (
...
 
 export const CancelListingIcon = (props: SVGProps) => (
-  <svg fill="none" xmlns="http://www.w3.org/2000/svg" {...props}>
+  <svg width="15" height="15" viewBox="0 0 15 15" fill="none" xmlns="http://www.w3.org/2000/svg" {...props}>
     <path
-      d="M71 31L75.36 35.36C76.85 36.8589 77.6863 38.8865 77.6863 41C77.6863 43.1135 76.85 45.1411 75.36 46.64L46.68 75.32C45.937 76.0638 45.0547 76.6539 44.0835 77.0565C43.1123 77.4591 42.0713 77.6663 41.02 77.6663C39.9687 77.6663 38.9277 77.4591 37.9565 77.0565C36.9853 76.6539 36.103 76.0638 35.36 75.32L31 71M47.8 7.8L41 1H1V41L7.8 47.8M77.6863 1L1.62987 77.0565M21 21H21.0333"
-      stroke="currentColor"
-      strokeWidth="2"
+      d="M12.6667 6L13.3933 6.72667C13.6417 6.97648 13.7811 7.31442 13.7811 7.66667C13.7811 8.01891 13.6417 8.35685 13.3933 8.60667L8.61333 13.3867C8.4895 13.5106 8.34245 13.609 8.18059 13.6761C8.01872 13.7432 7.84522 13.7777 7.67 13.7777C7.49478 13.7777 7.32128 13.7432 7.15941 13.6761C6.99755 13.609 6.8505 13.5106 6.72667 13.3867L6 12.6667M8.8 2.13333L7.66667 1H1V7.66667L2.13333 8.8M13.7811 1L1.10498 13.6761M4.33333 4.33333H4.33889"
+      stroke="#FD766B"
```
Comment (JackShort) [inline src/nft/components/icons.tsx:1501]: lets make this `currentColor` for ease of use

File: src/nft/hooks/useBag.ts (hunk 4 above)
Comment (JackShort) [reply src/nft/hooks/useBag.ts:63]: why is this necessary? you could just check if `bagExpanded` and if its not you `toggleBag`

File: src/nft/types/common/common.ts [hunk 10]
```diff
@@ -75,6 +75,14 @@ export interface Rarity {
   providers: { provider: string; rank: number; url?: string; score: number }[]
 }
 
+export interface Trait {
```
Comment (JackShort) [inline src/nft/types/common/common.ts:78]: seems like you can remove `OpenSeaUser` now

**Review by JackShort — CHANGES_REQUESTED** [review]

File: src/nft/components/details/AssetActivity.tsx [hunk 11]
```diff
@@ -102,7 +102,7 @@ const ActivityContainer = styled.div`
     width: 4px;
   }
   ::-webkit-scrollbar-track {
-    margin-top: 40px;
+    // margin-top: 40px;
```
Comment (JackShort) [inline src/nft/components/details/AssetActivity.tsx:105]: can we remove this comment?

File: src/nft/components/details/AssetDetails.tsx [hunk 12]
```diff
@@ -43,7 +43,7 @@ const OpacityTransition = css`
   }) => `opacity ${duration.medium} ${timing.ease}`};
 `
 
-const CollectionHeader = styled(RouterLink)`
+const CollectionHeader = styled(RouterLink)`x
```
Comment (JackShort) [inline src/nft/components/details/AssetDetails.tsx:46]: should there be an x here?

File: src/nft/components/details/DetailsContainer.tsx [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+          <Center onClick={copy}>
+            {shortenAddress(address, 2, 4)} <CopyIcon size={13} />
+          </Center>
+        }
+      />
+      <GridItem header="Token ID" body={tokenId.length > 9 ? stringShortener(tokenId) : tokenId} />
+      <GridItem header="Token standard" body={tokenType} />
+      <GridItem header="Blockchain" body="Ethereum" />
+      <GridItem header="Total supply" body={`${putCommas(totalSupply)}`} />
+      <GridItem header="Creator" body={shortenAddress(creator.address, 2, 4)} />
```
Comment (JackShort) [inline src/nft/components/details/DetailsContainer.tsx:95]: make the creator type on asset nullish and check if null - it crashes here for some collections

File: src/nft/components/details/AssetDetails.css.ts [hunk 14]
```diff
@@ -6,10 +6,9 @@ import { sprinkles, vars } from '../../css/sprinkles.css'
 export const image = style([
   sprinkles({ borderRadius: '20', height: 'full', alignSelf: 'center' }),
   {
-    width: 'calc(90vh - 165px)',
-    height: 'calc(90vh - 165px)',
-    maxHeight: '678px',
-    maxWidth: '678px',
+    maxHeight: 'calc(90vh - 165px)',
+    maxWidth: 780,
```
Comment (JackShort) [inline src/nft/components/details/AssetDetails.css.ts:10]: this needs to be '780px' and '400px'

**Comment by JackShort** [comment]
everything looks good with me I would like @cbachmeier to take a look after

File: src/nft/pages/collection/index.tsx [hunk 15]
```diff
@@ -129,7 +129,7 @@ const Collection = () => {
               </CollectionDescriptionSection>
               <CollectionDisplaySection>
                 <Box position="sticky" top="72" width="0">
-                  {isFiltersExpanded && <Filters traitsByGroup={collectionStats?.traits ?? {}} />}
+                  {/* {isFiltersExpanded && <Filters traitsByGroup={collectionStats?.traits ?? {}} />} */}
```
Comment (cbachmeier) [inline src/nft/pages/collection/index.tsx:132]: Re add filters?

File: src/graphql/data/nft/Details.ts [hunk 16]
```diff
@@ -141,15 +140,17 @@ export function useDetailsQuery(address: string, tokenId: string): [GenieAsset,
...
-        address: asset?.creator?.address,
+        profile_img_url: asset?.creator?.profileImage?.url ?? '',
+        address: asset?.creator?.address ?? '',
       },
-      metadataUrl: asset?.metadataUrl ?? undefined,
+      metadataUrl: asset?.metadataUrl ?? '',
       traits: asset?.traits?.map((trait) => {
-        return { trait_type: trait.name ?? undefined, trait_value: trait.value ?? undefined } as Trait
+        // @ts-ignore
+        return { trait_type: trait.name ?? '', value: trait.value ?? '' } as Trait
```
Comment (cbachmeier) [inline src/graphql/data/nft/Details.ts:151]: change `value:` to `trait_value:` and you can remove the ts-ignore

File: src/graphql/data/nft/Details.ts [hunk 17]
```diff
@@ -141,15 +140,17 @@ export function useDetailsQuery(address: string, tokenId: string): [GenieAsset,
...
       }),
+      externalLink: '',
```
Comment (cbachmeier) [inline src/graphql/data/nft/Details.ts:153]: externalLink has been replaced with metadataUrl

File: src/nft/components/details/AssetPriceDetails.tsx [hunk 18]
```diff
@@ -222,15 +311,37 @@ export const AssetPriceDetails = ({ asset, collection }: AssetPriceDetailsProps)
...
 
   return (
     <Container>
+      <OwnerInformationContainer>
+        <OwnerText
+          target="_blank"
+          href={`https://etherscan.io/address/${asset.owner.address}`}
+          rel="noopener noreferrer"
+        >
+          Seller:{' '}
```
Comment (cbachmeier) [inline src/nft/components/details/AssetPriceDetails.tsx:328]: Hide Seller on Pooled Assets (comes up blank)

File: src/nft/components/details/AssetPriceDetails.tsx [hunk 19]
```diff
@@ -190,26 +279,26 @@ export const NotForSale = ({ collection }: { collection: CollectionInfoForAsset
...
+const SubHeader = styled(ThemedText.SubHeader)`
+  color: ${({ theme }) => theme.textPrimary};
+`
+
 export const AssetPriceDetails = ({ asset, collection }: AssetPriceDetailsProps) => {
   const { account } = useWeb3React()
-  const cheapestOrder = asset.sellorders && asset.sellorders.length > 0 ? asset.sellorders[0] : undefined
-  const expirationDate = cheapestOrder
-    ? new Date((cheapestOrder as Deprecated_SellOrder).orderClosingDate ?? (cheapestOrder as SellOrder).endAt)
-    : undefined
```
Comment (cbachmeier) [inline src/nft/components/details/AssetPriceDetails.tsx:198]: expiration isn't showing for me and I believe it's the editing of this logic

File: src/nft/components/details/DetailsContainer.tsx [hunk 20]
```diff
@@ -0,0 +1,100 @@
...
+          <Center onClick={copy}>
+            {shortenAddress(address, 2, 4)} <CopyIcon size={13} />
+          </Center>
+        }
+      />
+      <GridItem header="Token ID" body={tokenId.length > 9 ? stringShortener(tokenId) : tokenId} />
+      <GridItem header="Token standard" body={tokenType} />
+      <GridItem header="Blockchain" body="Ethereum" />
+      <GridItem header="Total supply" body={`${putCommas(totalSupply ?? 0)}`} />
+      <GridItem header="Creator" body={creator?.address && shortenAddress(creator.address, 2, 4)} />
```
Comment (cbachmeier) [inline src/nft/components/details/DetailsContainer.tsx:95]: should this also be a link?

File: src/nft/components/details/TraitsContainer.tsx [hunk 21]
```diff
@@ -0,0 +1,99 @@
...
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  max-width: 100%;
+`
+
+const GridItem = ({ trait, collectionAddress }: { trait: Trait; collectionAddress: string }) => {
+  const { trait_type } = trait
+  // @ts-ignore
+  const trait_value = trait.trait_value || trait.value
```
Comment (cbachmeier) [inline src/nft/components/details/TraitsContainer.tsx:71]: if you implement my fix in src/graphql/data/nftDetails about renaming this param trait_value lns69-71 can be replaced with: `const { trait_type, trait_value } = trait`

File: src/nft/pages/asset/Asset.tsx [hunk 22]
```diff
@@ -37,6 +50,10 @@ const Asset = () => {
     [data, gqlData, isNftGraphQl]
   )
 
+  const { data: collectionStats } = useQuery(['collectionStats', contractAddress], () =>
```
Comment (cbachmeier) [inline src/nft/pages/asset/Asset.tsx:53]: Are there currently any fields missing from the collection object returned from GraphQL that collectionStats has? If not can remove this extra input and query?

**Review by cbachmeier — CHANGES_REQUESTED** [review]

File: src/nft/pages/asset/Asset.tsx (hunk 22 above)
Comment (aballerr) [reply src/nft/pages/asset/Asset.tsx:53]: the social links are not returned in the original query.  Not sure if that will change with graphql

File: src/nft/components/details/AssetPriceDetails.tsx (hunk 19 above)
Comment (aballerr) [reply src/nft/components/details/AssetPriceDetails.tsx:198]: some orders don't have a closing date.  I forget what shows up but this is intentional

File: src/nft/components/details/TraitsContainer.tsx (hunk 21 above)
Comment (aballerr) [reply src/nft/components/details/TraitsContainer.tsx:71]: this is to accommodate the v3 endpoint since it's at the ```value``` key

Comment (cbachmeier) [reply src/nft/components/details/TraitsContainer.tsx:71]: If you modify the v3 fetcher to cast the value to trait_value everything can use the Trait obj

File: src/graphql/data/nft/Details.ts (hunk 16 above)
Comment (cbachmeier) [reply src/graphql/data/nft/Details.ts:151]: Still needs to be added

File: src/nft/pages/asset/Asset.tsx (hunk 22 above)
Comment (cbachmeier) [reply src/nft/pages/asset/Asset.tsx:53]: it can be added to the gql query, can sync on that add after this PR.

**Review by cbachmeier — CHANGES_REQUESTED** [review]

**Review by cbachmeier — APPROVED** [review]
LGTM

**Pull request merged**